### PR TITLE
PR for fundamental core structure changes

### PR DESCRIPTION
**Pull request by jamesmcroft**
**Describe the changes made**
This PR contains fundamental changes to the MADE App Components libraries which are described below.

### Changes to minimum target version
Changes have been made to target the lowest minimum target version of .NET Standard and UWP so that the libraries can be installed with more projects.

This change sees the .NET Standard version lower from 2.0 to 1.4 and the UWP version to cover all versions instead of Fall Creators Update+

### Changes to incorporate XPlat
Other changes associated with this PR include adding support for XPlat where MADE was using custom built alternatives. These changes have been kept to a minimum but as a co-owner of the project, I suggest that we start incorporating XPlat as the base API layer for building controls and other functions in MADE.

Some examples include:

- MADE has a helper class for UI action dispatching which uses native APIs. XPlat contains a CoreDispatcher which does the same thing so has been replaced with it for consistency.

- MADE has a color, thickness and orientation class. XPlat contains alternatives which are cross-platform and have been replaced where used.

### Changes to NuGet package references

All package references have been updated to the latest versions. 

The UWP libraries have been downgraded to a supported version (6.1.9).

Included as part of this change has been to replace the older PCL variant of MvvmLight with the .NET Standard version. This will have a breaking change on users who are currently using this as the CommonServiceLocator is no longer used which is described here: http://www.mvvmlight.net/std10

There currently is an issue outstanding in the MvvmLight project that users who are implementing Android functionality and use the MvvmLight android support extension cannot use this. There is an active PR for this (https://github.com/lbugnion/mvvmlight/pull/25) but has been open since March last year. 

There is also an open PR from myself with MvvmLight regarding Android support packages references (https://github.com/lbugnion/mvvmlight/issues/60) with a PR (https://github.com/lbugnion/mvvmlight/pull/61) which have not yet been resolved.

@tom-made would it be wise if we forked MvvmLight and created our own internal packages to resolve these issues?

**What issues are associated?**
No associated issues

### CODE REVIEW

File: MADE.Samples.Android/MADE.Samples.Android.csproj [hunk 1]
```diff
@@ -126,12 +126,6 @@
     </ProjectReference>
   </ItemGroup>
   <ItemGroup>
-    <PackageReference Include="CommonServiceLocator">
```
[the remark this answers is not in the document]
Comment (jamesmcroft) [reply MADE.Samples.Android/MADE.Samples.Android.csproj]: Re-added the MvvmLight .NET standard library here. 

The XPlat one wasn't required.

File: MADE.App.Design/MADE.App.Design.csproj [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>netstandard2.0;xamarin.ios10;monoandroid81;uap10.0.16299</TargetFrameworks>
+    <TargetFrameworks>netstandard1.4;xamarin.ios10</TargetFrameworks>
```
Comment (tom-made) [inline MADE.App.Design/MADE.App.Design.csproj]: Why was the Android & UWP references removed here?

File: MADE.App.Mvvm.UnitTests/MADE.App.Mvvm.UnitTests.csproj [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
+    <TargetFramework>netcoreapp2.1</TargetFramework>
 
     <IsPackable>false</IsPackable>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.0.1" />
-    <PackageReference Include="MSTest.TestAdapter" Version="1.4.0" />
-    <PackageReference Include="MSTest.TestFramework" Version="1.4.0" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.9.0" />
```
Comment (tom-made) [inline MADE.App.Mvvm.UnitTests/MADE.App.Mvvm.UnitTests.csproj]: There was a downgrade on dependencies here, can we update them?

File: MADE.App.Views.Controls.HeaderedTextBlock.Forms/MADE.App.Views.Controls.HeaderedTextBlock.Forms.csproj [hunk 4]
```diff
@@ -25,6 +25,7 @@
   <ItemGroup>
     <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0-beta-63127-02" PrivateAssets="All" />
     <PackageReference Include="Xamarin.Forms" Version="3.6.0.220655" />
+    <PackageReference Include="XPlat.UI.Controls" Version="1.6.19073.2-alpha" />
```
Comment (tom-made) [inline MADE.App.Views.Controls.HeaderedTextBlock.Forms/MADE.App.Views.Controls.HeaderedTextBlock.Forms.csproj]: As we are using the pre-release versions of your XPlat packages, are we going to ensure that the official release is available before we get MADE to 1.0?

File: MADE.App.Design/MADE.App.Design.csproj (hunk 2 above)
Comment (jamesmcroft) [reply MADE.App.Design/MADE.App.Design.csproj]: At present, there is no component in here that requires platform specifics for Android and Windows so these were removed. This package can still be installed as part of an Android or Windows project through the .NET Standard 1.4 reference. We can re-add in the future if we need platform specifics for Windows and Android.

File: MADE.App.Mvvm.UnitTests/MADE.App.Mvvm.UnitTests.csproj (hunk 3 above)
Comment (jamesmcroft) [reply MADE.App.Mvvm.UnitTests/MADE.App.Mvvm.UnitTests.csproj]: Good catch. Will update these

File: MADE.App.Views.Controls.HeaderedTextBlock.Forms/MADE.App.Views.Controls.HeaderedTextBlock.Forms.csproj (hunk 4 above)
Comment (jamesmcroft) [reply MADE.App.Views.Controls.HeaderedTextBlock.Forms/MADE.App.Views.Controls.HeaderedTextBlock.Forms.csproj]: So I've created milestones for 1.6 of XPlat and 1.0 of MADE which should be available on the same day.

File: MADE.App.Mvvm.UnitTests/MADE.App.Mvvm.UnitTests.csproj (hunk 3 above)
Comment (jamesmcroft) [reply MADE.App.Mvvm.UnitTests/MADE.App.Mvvm.UnitTests.csproj]: This is now fixed

**Pull request merged**